tox.ini

```diff
@@ -106,3 +106,13 @@ depends =
 
 [coverage:run]
 relative_files = True
+
+[testenv:pack-{22.04,24.04}]
+description = Copy the corresponding charmcraft-{}.yaml to charmcraft.yaml and pack
+allowlist_externals =
+    sh
+commands =
+    # 22.04 is the default so we don't need to restore anything after
+    22.04: sh -c "ln -srf {toxinidir}/charmcraft-22.04.yaml {toxinidir}/charmcraft.yaml && charmcraft pack --project-dir={toxinidir}"
+    24.04: sh -c "ln -srf {toxinidir}/charmcraft-24.04.yaml {toxinidir}/charmcraft.yaml && charmcraft pack --project-dir={toxinidir}; ln -srf {toxinidir}/charmcraft-22.04.yaml {toxinidir}/charmcraft.yaml"
```

Comment:
What about making these do:
* backup current charmcraft.yaml
* make their own charmcraft.yaml pointing to the right thing
* pack
* restore charmcraft.yaml
?

It feels odd that `22.04` and `24.04` have different properties around their statefulness.  Agreed that it'll work atm, but this is an easy thing for someone to misuse without realizing.

Reply:
Not sure what's the benefit of this approach.
- If we symlink then if someone interrupts the 24.04 and `git add .` then we'd see a small diff (the symlink changed from 22.04 to 24.04).
- If we use a third copy of the yaml + untracked backup, then if someone `git add .` then we'd see a slightly larger diff (the charmcraft.yaml will have an entire section replaced).

A symlink seems cleaner. What am I missing?

Reply:
I'm not against a symlink, just the asymmetry between the two versions:
* 22.04 changes the charmcraft.yaml to the current case, packs
* 24.04 changes the charmcraft.yaml to the current case, packs, *and then back to a different case*

They don't feel equivalent enough to bundle together.  They'll work atm, but someone in future is going to miss that subtlety.  What I'm proposing is:

* X: backs up charmcraft.yaml, changes charmcraft.yaml to X, packs, restores backed up charmcraft.yaml

That way all versions of the environment work the same.

Reply:
An alternative would be have only a `pack-24.04` environment (no 22.04).  That feels more suitable imo, since the idea is packing for 24.04 is currently a special, manual case.

Reply:
hah, I see @lucabello asked about this [too](https://github.com/canonical/grafana-agent-operator/pull/174#discussion_r1734092453).  I like his original suggestion (or, doing the same thing but with backing up symlinks).

Reply:
Huh, yeah, might as well drop the 22.04 option. Thanks!